google/resource_compute_http_health_check.go

```diff
@@ -94,159 +109,260 @@ func resourceComputeHttpHealthCheckCreate(d *schema.ResourceData, meta interface
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
 	}
-	if v, ok := d.GetOk("port"); ok {
-		hchk.Port = int64(v.(int))
+
+	obj := map[string]interface{}{
+		"checkIntervalSec":   expandComputeHttpHealthCheckCheckIntervalSec(d.Get("check_interval_sec")),
+		"description":        expandComputeHttpHealthCheckDescription(d.Get("description")),
+		"healthyThreshold":   expandComputeHttpHealthCheckHealthyThreshold(d.Get("healthy_threshold")),
+		"host":               expandComputeHttpHealthCheckHost(d.Get("host")),
+		"name":               expandComputeHttpHealthCheckName(d.Get("name")),
+		"port":               expandComputeHttpHealthCheckPort(d.Get("port")),
+		"requestPath":        expandComputeHttpHealthCheckRequestPath(d.Get("request_path")),
+		"timeoutSec":         expandComputeHttpHealthCheckTimeoutSec(d.Get("timeout_sec")),
+		"unhealthyThreshold": expandComputeHttpHealthCheckUnhealthyThreshold(d.Get("unhealthy_threshold")),
 	}
-	if v, ok := d.GetOk("timeout_sec"); ok {
-		hchk.TimeoutSec = int64(v.(int))
+
+	url, err := replaceVars(d, config, "https://www.googleapis.com/compute/v1/projects/{{project}}/global/httpHealthChecks/{{name}}")
+	if err != nil {
+		return err
 	}
-	if v, ok := d.GetOk("unhealthy_threshold"); ok {
-		hchk.UnhealthyThreshold = int64(v.(int))
+
+	log.Printf("[DEBUG] Updating HttpHealthCheck %q: %#v", d.Id(), obj)
+	res, err := sendRequest(config, "PUT", url, obj)
+
+	if err != nil {
+		return fmt.Errorf("Error updating HttpHealthCheck %q: %s", d.Id(), err)
 	}
 
-	log.Printf("[DEBUG] HttpHealthCheck patch request: %#v", hchk)
-	op, err := config.clientCompute.HttpHealthChecks.Patch(
-		project, hchk.Name, hchk).Do()
+	op := &compute.Operation{}
+	err = Convert(res, op)
 	if err != nil {
-		return fmt.Errorf("Error patching HttpHealthCheck: %s", err)
+		return err
 	}
 
-	// It probably maybe worked, so store the ID now
-	d.SetId(hchk.Name)
+	err = computeOperationWaitTime(
+		config.clientCompute, op, project, "Updating HttpHealthCheck",
+		int(d.Timeout(schema.TimeoutUpdate).Minutes()))
 
-	err = computeOperationWait(config.clientCompute, op, project, "Updating Http Health Check")
 	if err != nil {
 		return err
 	}
 
 	return resourceComputeHttpHealthCheckRead(d, meta)
 }
 
-func resourceComputeHttpHealthCheckRead(d *schema.ResourceData, meta interface{}) error {
+func resourceComputeHttpHealthCheckDelete(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
 
 	project, err := getProject(d, config)
 	if err != nil {
 		return err
 	}
 
-	hchk, err := config.clientCompute.HttpHealthChecks.Get(
-		project, d.Id()).Do()
+	url, err := replaceVars(d, config, "https://www.googleapis.com/compute/v1/projects/{{project}}/global/httpHealthChecks/{{name}}")
 	if err != nil {
-		return handleNotFoundError(err, d, fmt.Sprintf("HTTP Health Check %q", d.Get("name").(string)))
+		return err
```

Comment:
We should update our magic module to handle 404 in delete properly

Reply:
We don't usually do 404 checking in deletes, since hypothetically the refresh would have caught it and removed the resource from state. Do you have an example resource that does?

Reply:
this resource, look at the diff

Reply:
It's diffing it against read() because the order of the functions changed

Reply:
:)